components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt

```diff
@@ -37,17 +39,117 @@ class GeckoEngine(
+            GeckoSession.NavigationDelegate.ERROR_CATEGORY_URI -> {
+                when (errorCode) {
+                    GeckoSession.NavigationDelegate.ERROR_UNKNOWN_HOST ->
+                        ErrorPages.ErrorType.ERROR_UNKNOWN_HOST
+                    GeckoSession.NavigationDelegate.ERROR_MALFORMED_URI ->
+                        ErrorPages.ErrorType.ERROR_MALFORMED_URI
+                    GeckoSession.NavigationDelegate.ERROR_UNKNOWN_PROTOCOL ->
+                        ErrorPages.ErrorType.ERROR_UNKNOWN_PROTOCOL
+                    GeckoSession.NavigationDelegate.ERROR_FILE_NOT_FOUND ->
+                        ErrorPages.ErrorType.ERROR_FILE_NOT_FOUND
+                    GeckoSession.NavigationDelegate.ERROR_FILE_ACCESS_DENIED ->
+                        ErrorPages.ErrorType.ERROR_FILE_ACCESS_DENIED
+                    else -> throw IllegalArgumentException("No error type definition for error code $errorCode")
+                }
+            }
+            GeckoSession.NavigationDelegate.ERROR_CATEGORY_PROXY -> {
+                when (errorCode) {
+                    GeckoSession.NavigationDelegate.ERROR_PROXY_CONNECTION_REFUSED ->
+                        ErrorPages.ErrorType.ERROR_PROXY_CONNECTION_REFUSED
+                    GeckoSession.NavigationDelegate.ERROR_UNKNOWN_PROXY_HOST ->
+                        ErrorPages.ErrorType.ERROR_UNKNOWN_PROXY_HOST
+                    else -> throw IllegalArgumentException("No error type definition for error code $errorCode")
+                }
+            }
+            GeckoSession.NavigationDelegate.ERROR_CATEGORY_SAFEBROWSING -> {
+                when (errorCode) {
+                    GeckoSession.NavigationDelegate.ERROR_SAFEBROWSING_MALWARE_URI ->
+                        ErrorPages.ErrorType.ERROR_SAFEBROWSING_MALWARE_URI
+                    GeckoSession.NavigationDelegate.ERROR_SAFEBROWSING_UNWANTED_URI ->
+                        ErrorPages.ErrorType.ERROR_SAFEBROWSING_UNWANTED_URI
+                    GeckoSession.NavigationDelegate.ERROR_SAFEBROWSING_HARMFUL_URI ->
+                        ErrorPages.ErrorType.ERROR_SAFEBROWSING_HARMFUL_URI
+                    GeckoSession.NavigationDelegate.ERROR_SAFEBROWSING_PHISHING_URI ->
+                        ErrorPages.ErrorType.ERROR_SAFEBROWSING_PHISHING_URI
+                    else -> throw IllegalArgumentException("No error type definition for error code $errorCode")
+                }
+            }
+            else -> throw IllegalArgumentException("No error type definition for error category $errorCategory")
+        }
+
+        throw IllegalArgumentException("No error type definition for error category $errorCategory")
+    }
+
     /**
      * See [Engine.settings]
      */
     override val settings: Settings = object : Settings {
         override var javascriptEnabled: Boolean
             get() = runtime.settings.javaScriptEnabled
-            set(value) { runtime.settings.javaScriptEnabled = value }
```

Comment:
Let's try not to format lines that we aren't touching. We're using ktlint's formatting rules which you can apply to your problem locally with `ktlint --apply-to-idea-project --android`.